collect_app/src/main/java/org/odk/collect/android/activities/SplashScreenActivity.java

```diff
@@ -231,7 +231,13 @@ public void run() {
     }
 
     private void initMapBox() {
-        MapView mapView = findViewById(R.id.mapView);
-        mapView.getMapAsync(mapBoxMap -> mapBoxMap.setStyle(Style.MAPBOX_STREETS, style -> { }));
+        // This "one weird trick" lets us initialize MapBox at app start when the internet is
+        // most likely to be available. This is annoyingly needed for offline tiles to work.
+        try {
+            MapView mapView = getLayoutInflater().inflate(R.layout.mapbox_init, null).findViewById(R.id.mapView);
```

Comment:
Are you sure this still does the job? I mean the map is not displayed anywhere but it might be required.

Reply:
Not had a chance to double check yet. I could see that being a problem.

Reply:
@grzesiek2010 if you have basemaps already ready to go could you check? I don't have anything setup for any of this.

Reply:
Ok will do.

Reply:
Thanks!

Reply:
Yeah as I thought it doesn't work. It needs to be displayed.

Reply:
Ach really? That's weird. I wondewr what part of the View's lifecycle needs to get hit. Definitely possible to inflate and then add this into the view hierarchy though. I'm not going to be able to look at this again tonight though. @lognaturel if this is getting in your way (i.e. you need to look at the app on an emulator) I'd suggest merging this or reverting #3804. Also feel free to push to this if you have a version that retains the fixes from #3804!

Reply:
I'll take a look.

Reply:
Ok it's done. I moved the code from the SplashScreen to the MainMenuActivity because once we merge https://github.com/getodk/collect/pull/3813 the SplashScreen won't be displayed by default so the fix also won't work.